curvefs/src/client/s3/client_s3_adaptor.cpp

```diff
@@ -105,10 +107,19 @@ int S3ClientAdaptorImpl::Write(uint64_t inodeId, uint64_t offset,
 
     FileCacheManagerPtr fileCacheManager =
         fsCacheManager_->FindOrCreateFileCacheManager(fsId_, inodeId);
-    if (fsCacheManager_->WriteCacheIsFull()) {
+    ioMtx_.lock();
+    pendingReq_.fetch_add(1, std::memory_order_seq_cst);
+    VLOG(6) << "pendingReq_ is: " << pendingReq_;
+    uint64_t pendingReq = pendingReq_.load(std::memory_order_seq_cst);
+    if ((fsCacheManager_->GetDataCache() + pendingReq*fuseMaxSize_)
+      >= fsCacheManager_->GetDataCacheMax()) {
         LOG(INFO) << "write cache is full, wait flush";
+        fsCacheManager_->DataCacheByteInc(length);
```

Comment:
Why we call DataCacheByteInc here when this request is pendding?

Reply:
all pending req will calculate prevent from mem cache exceed s3.writeCacheMaxByte